src/viam/module/module.py

```diff
@@ -64,6 +64,16 @@ def from_args(cls) -> Self:
         return cls(address, log_level=log_level)
 
     def __init__(self, address: str, *, log_level: int = logging.INFO) -> None:
+        if 'reconfigure' in dir(sys.stdout):
+            """Modules are launched from the viam-server, and a module's stdout is not connected to
+            a tty. Python reacts to that environment by disabling line buffering. Which means
+            `print` statements are not immediately flushed to the viam-server. This is confusing for
+            customers and can interfere with debugging. We reconfigure stdout and stderr here to
+            better align python with other languages/viam SDKs and user expectations.
+
+            Note: `reconfigure` only exists in python3.7+. """
```

Comment:
Could we make this docstring slightly less internal sounding and more for devs/users who might go poking around?

```
When a module is launched by viam-server, its stdout is not connected to a tty. 
In response, python desiables line buffering, which prevents `print` statements 
from being immediately flushed to viam-server. This behavior can be confusing, 
interfere with debugging, and is non-standard when compared to other languages. 
Here, stdout and stderr are reconfigured to immediately flush.
```
I think we can drop the note since we only support python3.8+ anyway

Reply:
I wasn't aware of the python minimum. Should I remove the `dir` check then as well? The existence of that note was to justify the conditional `if 'reconfigure' in dir(sys.stdout)` bit.

Reply:
Also I'm not sure what this CI error is saying:
![image](https://github.com/viamrobotics/viam-python-sdk/assets/71458/04c0e9d2-70b7-4f15-a47f-e807d83e7b92)

`stdout` and `stderr` are* of a TextIO[Wrapper] type. And the [API is published](https://docs.python.org/3/library/io.html#io.TextIOWrapper.reconfigure). Any guidance on if there's actually a problem here?

Reply:
Yup you can remove the `if reconfig` check.

Also the CI error is about this:

```
# TextIO is used instead of more specific types for the standard streams,
# since they are often monkeypatched at runtime. At startup, the objects
# are initialized to instances of TextIOWrapper.
#
# To use methods from TextIOWrapper, use an isinstance check to ensure that
# the streams have not been overridden:
#
# if isinstance(sys.stdout, io.TextIOWrapper):
#    sys.stdout.reconfigure(...)
```

Easily fixable